src/Development/IDE/Spans/Calculate.hs

```diff
@@ -98,8 +98,9 @@ getTypeLHsBind :: (GhcMonad m)
                => TypecheckedModule
                -> LHsBind GhcTc
                -> m [(SpanSource, SrcSpan, Maybe Type)]
-getTypeLHsBind _ (L _spn FunBind{fun_id = pid,fun_matches = MG{}}) =
-  return [(Named $ getName (unLoc pid), getLoc pid, Just (varType (unLoc pid)))]
+getTypeLHsBind _ (L _spn FunBind{ fun_id = pid
+                                , fun_matches = MG{mg_alts=(L _ matches)}}) =
+  return [(Named (getName (unLoc pid)), getLoc match, Just (varType (unLoc pid))) | match <- matches ]
```

Comment:
It looks like we no longer record the  location of `pid`. Is that guaranteed to be one of the matches?

Reply:
I don't have a language-lawyer-level guarantee, but I think that the location of the first match is the same as the location of the overall `pid`. Experimental evidence in support of this: before this fix, gotoDef only worked for the first clause, and jumped to itself; now gotoDef works for all clauses, and jumps to the first one.

Is that enough, or would you like me to dig a bit more?

Reply:
Definitely good enough, if it turns out to be a problem we can always address it later. Thanks for the response!